### PR TITLE
Added AgencyAccountsController for admin. Aded endpoints for get agen…

### PR DESCRIPTION
**Pull request by AlexeyShok**
Added AgencyAccountsController for admins. Added endpoints for get agency accounts and for put is active status for agency account.

### CODE REVIEW

File: Api/Controllers/AdministratorControllers/AgencyAccountsController.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        [HttpGet("{agencyId}/agency-accounts")]
+        [ProducesResponseType(typeof(List<AgencyAccountFullInfo>), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ProblemDetails), (int)HttpStatusCode.BadRequest)]
+        [AdministratorPermissions(AdministratorPermissions.CounterpartyBalanceObservation)]
+        public async Task<IActionResult> GetAgencyAccounts([FromRoute] int agencyId) 
+            => Ok(await _accountPaymentService.GetAgencyAccounts(agencyId));
+
+
+        /// <summary>
+        /// Sets agency account is active status
```
Comment (kirillta) [inline Api/Controllers/AdministratorControllers/AgencyAccountsController.cs:39]: I suggest `Changes an agency account activity state` or something

File: Api/Models/Agencies/AgencyAccountFullInfo.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+using HappyTravel.Money.Enums;
+using HappyTravel.Money.Models;
+using System;
+
+namespace HappyTravel.Edo.Api.Models.Agencies
+{
+    public readonly struct AgencyAccountFullInfo
```
Comment (kirillta) [inline Api/Models/Agencies/AgencyAccountFullInfo.cs:7]: `FullAgencyAccountInfo`. Lets keep more natural language

File: Api/Models/Agencies/AgencyAccountSettings.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿using Newtonsoft.Json;
+
+namespace HappyTravel.Edo.Api.Models.Agencies
+{
+    public readonly struct AgencyAccountSettings
+    {
+        [JsonConstructor]
+        public AgencyAccountSettings(bool isActive)
```
Comment (kirillta) [inline Api/Models/Agencies/AgencyAccountSettings.cs:8]: I suggest to add an agency ID as well

Comment (kirillta) [reply Api/Models/Agencies/AgencyAccountSettings.cs:8]: Or an agency ID and an account ID

File: Api/Controllers/AdministratorControllers/AgencyAccountsController.cs (hunk 1 above)
Comment (AlexeyShok) [reply Api/Controllers/AdministratorControllers/AgencyAccountsController.cs:39]: Fixed

File: Api/Models/Agencies/AgencyAccountFullInfo.cs (hunk 2 above)
Comment (AlexeyShok) [reply Api/Models/Agencies/AgencyAccountFullInfo.cs:7]: Fixed

File: Api/Models/Agencies/AgencyAccountSettings.cs (hunk 3 above)
Comment (AlexeyShok) [reply Api/Models/Agencies/AgencyAccountSettings.cs:8]: This model is only used to pass data in a request. Id agencies and accounts transferred in the route. Therefore, you do not need to add them to the model.

Comment (kirillta) [reply Api/Models/Agencies/AgencyAccountSettings.cs:8]: I see the opposite: 
```csharp
await _accountPaymentService.SetAgencyAccountSettings(agencyId, agencyAccountId, agencyAccountSettings)
```

**Review by kirillta — CHANGES_REQUESTED** [review]

**Review by kirillta — APPROVED** [review]

**Pull request merged**